tutor/src/models/course.ts

```diff
@@ -95,19 +93,19 @@ export default class Course extends BaseModel {
     @lazyGetter gradingTemplates = new GradingTemplates({ course: this });
     @lazyGetter practiceQuestions = new PracticeQuestions({ course: this });
 
-    @hasMany({ model: Period, inverseOf: 'course', extend: getters({
+    @model(Period) periods = []; /* extend: getters({
         sorted() { return PH.sort(this.active);                        },
         archived() { return filter(this, period => !period.is_archived); },
         active() { return filter(this, 'isActive'); },
-    }) }) periods = [];
+    }) */
 
-    @hasMany({ model: Role, inverseOf: 'course', extend: getters({
+    @model(Role) roles; /* extend: getters({
         student() { return find(this, { isStudent: true }); },
         teacher() { return find(this, { isTeacher: true }); },
         teacherStudent() { return find(this, { isTeacherStudent: true }); },
-    }) }) roles;
-    @hasMany({ model: Student, inverseOf: 'course' }) students;
-    @hasMany({ model: TeacherProfiles, inverseOf: 'course' }) teacher_profiles;
+    }) */
+    @model(Student) students;
```

Comment:
hm, hasMany needs to be set to an empty array, that's how the model code knows how it should be treated.  So this should become `@model(Student) students = []`

You can also omit `;` if it's easier, we no longer require them